bundles/org.openhab.binding.omnikinverter/src/main/java/org/openhab/binding/omnikinverter/internal/OmnikInverterHandlerFactory.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.omnikinverter.internal;
+
+import static org.openhab.binding.omnikinverter.OmnikInverterBindingConstants.THING_TYPE_OMNIK;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.omnikinverter.handler.OmnikInverterHandler;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link OmnikInverterHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Hans van den Bogert - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.omnikinverter", service = ThingHandlerFactory.class)
+public class OmnikInverterHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_OMNIK);
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_OMNIK.equals(thingTypeUID)) {
```

Comment:
I think that was part of the generated skeleton at one point? If I ever need to add more thing-types, I'd have to refactor back to the current situation I guess, since the ternary wouldn't help that much. Do you feel this is really needed?